### PR TITLE
Add async iterator sample to the SignalR streaming doc

### LINKED ISSUE #11129: Add async iterator sample to the SignalR streaming doc
See https://github.com/aspnet/AspNetCore/pull/6791

Write an article explaining how to return IAsyncEnumerable responses from server Hub methods. The article should include:

* [x] Async generators
* [x] Cancellation

### PR DESCRIPTION
**Pull request by halter73**
Fixes #11129

### CODE REVIEW

File: aspnetcore/signalr/streaming.md [hunk 1]
```diff
@@ -19,10 +19,20 @@ ASP.NET Core SignalR supports streaming return values of server methods. This is
 
 ## Set up the hub
 
-A hub method automatically becomes a streaming hub method when it returns a `ChannelReader<T>` or a `Task<ChannelReader<T>>`. Below is a sample that shows the basics of streaming data to the client. Whenever an object is written to the `ChannelReader` that object is immediately sent to the client. At the end, the `ChannelReader` is completed to tell the client the stream is closed.
+::: moniker range=">= aspnetcore-3.0"
+
+A hub method automatically becomes a streaming hub method when it returns a `ChannelReader<T>`, a `IAsyncEnumerable<T>`, `Task<ChannelReader<T>>` or `Task<IAsyncEnumerable<T>>`.
```
Comment (BrennanConroy) [inline aspnetcore/signalr/streaming.md]: remove "a" from in front of `IAsyncEnumerable<T>`

File: aspnetcore/signalr/streaming.md [hunk 2]
```diff
@@ -19,10 +19,20 @@ ASP.NET Core SignalR supports streaming return values of server methods. This is
...
+
+::: moniker-end
+
+::: moniker range="< aspnetcore-3.0"
+A hub method automatically becomes a streaming hub method when it returns a `ChannelReader<T>` or a `Task<ChannelReader<T>>`.
```
Comment (BrennanConroy) [inline aspnetcore/signalr/streaming.md]: nit: spacing around the moniker stuff

File: aspnetcore/signalr/streaming.md (hunk 1 above)
Comment (scottaddie) [inline aspnetcore/signalr/streaming.md]:
```suggestion
A hub method automatically becomes a streaming hub method when it returns a `ChannelReader<T>`, `IAsyncEnumerable<T>`, `Task<ChannelReader<T>>`, or `Task<IAsyncEnumerable<T>>`.
```

File: aspnetcore/signalr/streaming.md [hunk 3]
```diff
@@ -19,10 +19,20 @@ ASP.NET Core SignalR supports streaming return values of server methods. This is
...
+::: moniker-end
+
+Below is a sample that shows the basics of streaming data to the client using Channels. Whenever an object is written to the `ChannelWriter` that object is immediately sent to the client. At the end, the `ChannelWriter` is completed to tell the client the stream is closed.
```
Comment (scottaddie) [inline aspnetcore/signalr/streaming.md]:
```suggestion
The following sample shows the basics of streaming data to the client using Channels. Whenever an object is written to the `ChannelWriter` that object is immediately sent to the client. At the end, the `ChannelWriter` is completed to tell the client the stream is closed.
```

File: aspnetcore/signalr/streaming.md (hunk 1 above)
Comment (halter73) [reply aspnetcore/signalr/streaming.md]: Do we typically use the oxford comma?

File: aspnetcore/includes/csharp-8-required.md [hunk 4]
```diff
@@ -0,0 +1,2 @@
+> [!NOTE]
+> The sample below requires C# 8.0 or later.
```
Comment (halter73) [inline aspnetcore/includes/csharp-8-required.md]: @scottaddie I added csharp-8-required.md to the includes directory. This wasn't here when you first approved the PR. Does this look OK to you? Thanks.

File: aspnetcore/signalr/streaming.md [hunk 5]
```diff
@@ -1,7 +1,7 @@
 ---
 title: Use streaming in ASP.NET Core SignalR
 author: bradygaster
-description: 
+description:
```
Comment (scottaddie) [inline aspnetcore/signalr/streaming.md]: Add a brief, 1-sentence description of the doc

File: aspnetcore/includes/csharp-8-required.md (hunk 4 above)
Comment (scottaddie) [inline aspnetcore/includes/csharp-8-required.md]:
```suggestion
> The following sample requires C# 8.0 or later.
```

**Pull request merged**